packages/spanner/src/transaction.js

```diff
@@ -223,25 +228,26 @@ Transaction.prototype.begin = function(callback) {
  * });
  *
  * //-
- * // If the callback is omitted, we'll return a Promise.
+ * // To use promises, simply return a Promise.
  * //-
- * database.runTransaction()
- *   .then(function(data) {
- *     var transaction = data[0];
+ * database.runTransaction(function(err, transaction) {
+ *   if (err) {
+ *     // Error handling omitted.
```

Comment:
Just link-dropping-- posted the question to Twitter and asked around for advice. Dave and I agreed this sums up our two best options:

https://twitter.com/stephenplusplus/status/846822914490073088
https://gist.github.com/stephenplusplus/3cb49529ee537c640a4670edf900e294